src/components/inputs/rich-text-input/editor.js

```diff
@@ -116,72 +29,7 @@ const Editor = props => {
     if (!doesExceedCollapsedHeightLimit && renderToggleButton) {
       setRenderToggleButton(false);
     }
-  }, [props.editor.value, renderToggleButton]);
-
-  const activeBlock =
-    (props.editor.value.blocks.first() &&
-      props.editor.value.blocks.first().type) ||
-    '';
-
-  const activeMarks = Array.from(props.editor.value.activeMarks).map(
-    mark => mark.type
-  );
-
-  const activeMoreStyleMarks = activeMarks.filter(activeMark =>
-    dropdownOptions.some(dropdownOption => activeMark === dropdownOption.value)
-  );
-
-  const hasBlock = type =>
-    props.editor.value.blocks.some(node => node.type === type);
-
-  const onClickBlock = ({ value: type }) => {
-    // Handle everything but list buttons.
-    if (type !== BULLETED_LIST && type !== NUMBERED_LIST) {
-      const isActive = hasBlock(type);
-      const isList = hasBlock(LIST_ITEM);
-
-      if (isList) {
-        props.editor
-          .setBlocks(isActive ? DEFAULT_NODE : type)
-          .unwrapBlock(BULLETED_LIST)
-          .unwrapBlock(NUMBERED_LIST);
-      } else {
-        props.editor.setBlocks(isActive ? DEFAULT_NODE : type);
-      }
-    } else {
-      // Handle the extra wrapping required for list buttons.
-      const isList = hasBlock(LIST_ITEM);
-      const isType = props.editor.value.blocks.some(block => {
-        return !!props.editor.value.document.getClosest(
-          block.key,
-          parent => parent.type === type
-        );
-      });
-
-      if (isList && isType) {
-        props.editor
-          .setBlocks(DEFAULT_NODE)
-          .unwrapBlock(BULLETED_LIST)
-          .unwrapBlock(NUMBERED_LIST);
-      } else if (isList) {
-        props.editor
-          .unwrapBlock(type === BULLETED_LIST ? NUMBERED_LIST : BULLETED_LIST)
-          .wrapBlock(type);
-      } else {
-        props.editor.setBlocks(LIST_ITEM).wrapBlock(type);
-      }
-    }
-  };
-
-  const onChangeMoreStyles = val => {
-    if (!props.editor.value.selection.isFocused) {
-      props.editor.focus();
-    }
-    props.editor.toggleMark(val.value);
-  };
-
-  const hasUndos = props.editor.hasUndos();
-  const hasRedos = props.editor.hasRedos();
+  }, [props.editor.value.document, renderToggleButton]);
```

Comment:
this is a slight change -> now we only rerun the effect when the document changes, read about this in the slate JS slack channel.